### DISCUSSION TITLE
rt = doc->render(max_width) ?

### DISCUSSION THREAD

**Discussion by octalflow** [litehtml/litehtml#377] [Q&A]
Hi.

When I use the render function, the function return greater size.

int rt = doc->render(max_width);
if(rt > max_width) {
// HOW TO set my window size ?
}

**Comment by tordex** [comment]
First, ```document::render``` returns "best fit width". You can use this width to make window smaller/greater to fit content by width. But you need call ```document::render``` again with "best fit width" as argument.

The actual content width/height you can get with ```doc->content_width()``` and ```doc->content_height()``` functions.
Yes, the actual content width can be greater then ```max_width``` argument. This means litehtml can't make content smaller.

**Reply by octalflow** [reply]
thank you
